### PR TITLE
Add `bundle lock --add-checksums` to add checksums to an existing lockfile

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
## What was the end-user or developer problem that led to this PR?

There's no easy way to enable the lockfile checksums feature.

## What is your fix for the problem, implemented in this PR?

I plan to add two ways to opt in to the feature:

* A setting `lockfile_checksums` that controls whether new lockfiles include checksums or not.
* An `--add-checksums` flag to `bundle lock` to add checksums to an existing lockfile.

This PR implements the latter.

## Make sure the following tasks are checked

- [x] Describe the problem / feature
- [x] Write [tests](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#tests) for features and bug fixes
- [x] Write code to solve the problem
- [x] Make sure you follow the [current code style](https://github.com/rubygems/rubygems/blob/master/doc/bundler/development/PULL_REQUESTS.md#code-formatting) and [write meaningful commit messages without tags](https://github.com/rubygems/rubygems/blob/master/doc/bundler/development/PULL_REQUESTS.md#commit-messages)

### CODE REVIEW

**Review by simi — APPROVED** [review]

**Comment by miry** [comment]
```
$ bundle lock --add-checksums
Unknown switches "--add-checksums"

$ bundle version
Bundler version 2.6.1 (2024-12-17 commit 00a344e02c8)
```

**Comment by simi** [comment]
> ```
> $ bundle lock --add-checksums
> Unknown switches "--add-checksums"
> 
> $ bundle version
> Bundler version 2.6.1 (2024-12-17 commit 00a344e02c8)
> ```

@miry Seems your bundle is locked to other Bundler version (check BUNDLED WITH at the end of Gemfile.lock) and bundler switches to that version (missing checksums) automatically. Can you try  `bundle update --bundler` first to move whole bundle to Bundler 2.6.1 and add checksums after?

**Comment by miry** [comment]
@simi Actually I did `bundle update --bundler`, but i should first upgrade bundle in the gems with `gem install bundler`. Thanks for the tip.

TLDR: Worked for me:

```shell
$ gem install bundler
$ bundle update --bundler
$ bundle lock --add-checksums
Writing lockfile to .../Gemfile.lock
```

**Comment by miry** [comment]
Can you share the example of CHECKSUM lines?

**Comment by simi** [comment]
@miry thanks, I'll update the guide to make it clear.

Here's an example - https://github.com/rubygems/rubygems.org/blob/03c1225610e4cd759038409e802e5701000c2595/Gemfile.lock#L999.
